ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -172,8 +174,12 @@ protected void onSelectionChanged(int selStart, int selEnd) {
   protected void onFocusChanged(
       boolean focused, int direction, Rect previouslyFocusedRect) {
     super.onFocusChanged(focused, direction, previouslyFocusedRect);
-    if (focused && mSelectionWatcher != null) {
-      mSelectionWatcher.onSelectionChanged(getSelectionStart(), getSelectionEnd());
+    if (focused) {
+      if (mClearTextOnFocus) {
+        setText("");
```

Comment:
I did not test if onChange is fired and forgot to test controlled TextInputs behavior, will do that. Thanks for the suggestion!